libdrgn/error.c

```diff
@@ -117,6 +120,37 @@ LIBDRGN_PUBLIC struct drgn_error *drgn_error_format(enum drgn_error_code code,
 	return drgn_error_create_nodup(code, message);
 }
 
+LIBDRGN_PUBLIC struct drgn_error *drgn_error_create_fault(const char *message,
+							  uint64_t address)
+{
+	struct drgn_error *err;
+	int ret;
+
+	err = drgn_error_create(DRGN_ERROR_FAULT, message);
+	if (!err)
+		return err;
+	err->address = address;
```

Comment:
One last thing I noticed: `drgn_error_create()` never returns `NULL`, but it may return `&drgn_enomem`. This should be

```c
if (err != &drgn_enomem)
	err->address = address;
```